docs/resources/aws_kms_key.md.erb

```diff
@@ -0,0 +1,171 @@
+<br>
+
+## Syntax
+
+An aws_kms_key resource block identifies a key by key_arn or the key id.
+
+    # Find a kms key by arn
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+    
+    # Find a kms key by just the id
+    describe aws_kms_key('4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+
+    # Hash syntax for key arn
+    describe aws_kms_key(key_arn: 'arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+
+<br>
+
+## Examples
+
+The following examples show how to use this InSpec audit resource.
+
+### Test that the specified key does exist
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should exist }
+    end
+
+### Test that the specified key is enabled
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should be_enabled }
+    end
+
+### Test that the specified key is rotation enabled
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      it { should have_rotation_enabled }
+    end
+
+<br>
+
+## Properties
+
+### key\_id
+
+The globally unique identifier for the key.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('key_id') { should cmp 'arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321' }
+    end
+
+### arn
+
+The ARN identifier of the specified key. An ARN uniquely identifies the key within AWS.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('arn') { should cmp "arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321" }
+    end
+
+### creation_date
+
+Specifies the date and time when the key was created. 
+
+    # Makes sure that the key was created at least 10 days ago
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('creation_date') { should be < Time.now - 10 * 86400 }
+    end
+
+### created\_days\_ago
+
+Specifies the number of days since the key was created.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('created_days_ago') { should be > 10 }
+    end
+
+
+### key\_state
+
+Specifies the state of the key one of "Enabled", "Disabled", "PendingDeletion", "PendingImport". To just check if the key is enabled or not, use the be_enabled matcher.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('key_state') { should cmp "Enabled" }
+    end
+
+### description
+
+Specifies the description of the key.
+
+    describe aws_kms_key('arn:aws:kms:us-east-1::key/4321dcba-21io-23de-85he-ab0987654321') do
+      its('description') { should cmp "key-description" }
+    end
+
+### deletion\_date
```

Comment:
I think it's confusing (or at least irregular) that two properties that are DateTime-valued have different suffixes - `deletion_date` and `valid_to`.   While that is how the AWS API works, I think having a clearer interface is better.  

How about `deletion_time` instead of `deletion_date`?